css-animations-2/Overview.bs

```diff
@@ -847,6 +873,271 @@ expressed in milliseconds, it must be divided by 1,000 to produce a value in
 seconds before being assigned to the {{AnimationEvent/elapsedTime}} member of
 the {{AnimationEvent}}.
 
+# Animation Triggers # {#triggers}
+
+An <dfn export>animation trigger</dfn> is used to control the playback
+of its associated [=animation=] for time-driven animations.
+[=Animation triggers=] follow the same [[web-animations-1#hierarchical|hierarchy]]
+as animations and are associated with a [=timeline=] and an [=animation trigger effect|effect=].
+
+Issue: Should there be any effect of triggers on scroll-driven animations?
+
+##  Animation Trigger Effects ## {#trigger-effects}
+
+The <dfn>animation trigger effect</dfn> applies an effect
+to an [=animation=]’s playback via a mutually associated
+[=animation trigger=]. This effect is defined by its [=animation trigger type|type=].
+
+An [=animation trigger effect=] of [=animation trigger=] |trigger|
+with [=animation trigger type=] |type| has an internal
+<dfn export lt="animation-trigger-effect-state">state</dfn> that
+can be one of the following:
+
+<dl dfn-for=animation-trigger-effect-state dfn-type=value>
+  <dt><dfn>idle</dfn>
+  <dd>
+    The [=animation effect=] associated with |trigger| remains in
+    its [=animation effect/before phase=] and stays at zero [=animation/start time=].
+
+  <dt><dfn>primary</dfn>
+  <dd>
+    The ''primary'' effect defined by |type| is applied to its associated [=animation=].
+
+  <dt><dfn>inverse</dfn>
+  <dd>
+    The ''inverse'' effect defined by the |type| is applied to its associated [=animation=].
+
+</dl>
+
+Issue: Do we need a formal resolution on the spec of the idle state?
+
+An [=animation trigger effect=] |effect| is controlled
+by the [=animation attachment range|attachment range=] it is attached to,
+given a flag |did trigger|, as follows:
+
+<dl class=switch>
+  : If |effect|’s [=local time=] is [=unresolved=],
+  ::
+    It is in its ''animation-trigger-effect-state/idle'' state, and its |did trigger| flag is set to false.
+
+  : Otherwise,
+  ::
+
+    <dl class=switch>
+      : If |effect| is inside its [=active interval=],
```

Comment:
I think defining a trigger in terms of an internal animation effect's time might be overkill. E.g. I think we could define this simply in terms of the animation timeline's time with respect to the specified range

Reply:
I struggled with that quite a bit. Eventually it worked out once I defined the same hierarchy as animations', and defined a special state for the `effect`. I'll try simplifying this further.